vertx-pg-client/src/main/java/io/vertx/pgclient/PgPool.java

```diff
@@ -97,14 +98,31 @@ static PgPool pool(PgConnectOptions connectOptions, PoolOptions poolOptions) {
       vertxOptions.setPreferNativeTransport(true);
     }
     VertxInternal vertx = (VertxInternal) Vertx.vertx(vertxOptions);
-    return PgPoolImpl.create(vertx.getOrCreateContext(), true, connectOptions, poolOptions);
+    return PgPoolImpl.create(vertx.getOrCreateContext(), true, false, connectOptions, poolOptions);
   }
 
   /**
    * Like {@link #pool(PgConnectOptions, PoolOptions)} with a specific {@link Vertx} instance.
    */
   static PgPool pool(Vertx vertx, PgConnectOptions connectOptions, PoolOptions poolOptions) {
-    return PgPoolImpl.create(((VertxInternal)vertx).getOrCreateContext(), false, connectOptions, poolOptions);
+    return PgPoolImpl.create(((VertxInternal)vertx).getOrCreateContext(), false, false, connectOptions, poolOptions);
   }
 
+  /**
+   * Create a client backed by a connection pool to the database configured with the given {@code connectOptions} and {@code poolOptions}.
+   *
+   * @param poolOptions the options for creating the backing pool
+   * @return the client
+   */
+  static SqlClient client(PgConnectOptions connectOptions, PoolOptions poolOptions) {
```

Comment:
we need an overload factory method with vertx instance parameter.

Reply:
yes indeed